extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceBuildTimeConfig.java

```diff
@@ -32,5 +31,5 @@ public static final FlywayDataSourceBuildTimeConfig defaultConfig() {
      * scanned recursively down non-hidden directories.
      */
     @ConfigItem(defaultValue = DEFAULT_LOCATION)
-    public List<String> locations;
+    public Optional<String[]> locations;
```

Comment:
Why is it Optional? Since we have a default value, it shouldn't be.

Reply:
Any property that can be empty has to be optional, regardless of whether the default is non-empty.  If the property cannot be empty, then it should not be optional.